parquet-column/src/main/java/org/apache/parquet/column/values/bloomfilter/BlockSplitBloomFilter.java

```diff
@@ -199,27 +231,31 @@ public void writeTo(OutputStream out) throws IOException {
 
   @Override
   public void insertHash(long hash) {
-    int bucketIndex = (int)(hash >> 32) & (bitset.length / BYTES_PER_BLOCK - 1);
+    long numBlocks = bitset.length / BYTES_PER_BLOCK;
+    long lowHash = hash >>> 32;
+    int blockIndex = (int)(lowHash * numBlocks >> 32);
```

Comment:
What happens if this product overflows? How does that behavior compare to this line operating on unsigned values in C++, which cannot overflow on multiplication?

Reply:
The number of blocks right shift 5 bits at first, so its value should less than `2<<27` and the overflow should not happen here.